### PR TITLE
Change demo-base container build from conda to mamba

### PR DESCRIPTION
**Pull request by lresende**

### CODE REVIEW

File: etc/docker/demo-base/Dockerfile [hunk 1]
```diff
@@ -66,7 +66,8 @@ RUN echo "en_US.UTF-8 UTF-8" > /etc/locale.gen && \
 ADD fix-permissions /usr/local/bin/fix-permissions
 # Create jovyan user with UID=1000 and in the 'users' group
 # and make sure these dirs are writable by the `users` group.
-RUN groupadd wheel -g 11 && \
+RUN userdel -r mambauser && \
```
Comment (kevin-bates) [inline etc/docker/demo-base/Dockerfile:69]: Why is this necessary?  I wonder if that's part of my build issue.

**Review by kevin-bates — COMMENTED** [review]
For whatever reason, I'm unable to build a `demo-base:3.2.1` image locally.  It fails with the following:
```
Error in download.file(url = "https://cran.r-project.org/src/contrib/Archive/SparkR/SparkR_$SPARKR_VERSION.tar.gz",  : 
  cannot open URL 'https://cran.r-project.org/src/contrib/Archive/SparkR/SparkR_$SPARKR_VERSION.tar.gz'
In addition: Warning message:
In download.file(url = "https://cran.r-project.org/src/contrib/Archive/SparkR/SparkR_$SPARKR_VERSION.tar.gz",  :
  cannot open URL 'https://cran.r-project.org/src/contrib/Archive/SparkR/SparkR_$SPARKR_VERSION.tar.gz': HTTP status was '404 Not Found'
Execution halted
The command '/usr/local/bin/_dockerfile_shell.sh Rscript -e 'install.packages("IRkernel", repos="https://mirror.las.iastate.edu/CRAN/", lib="/opt/conda/lib/R/library")'             -e 'IRkernel::installspec(prefix = "/usr/local")'             -e 'download.file(url = "https://cran.r-project.org/src/contrib/Archive/SparkR/SparkR_$SPARKR_VERSION.tar.gz", destfile = "SparkR_$SPARKR_VERSION.tar.gz")'             -e 'install.packages(pkgs="SparkR_$SPARKR_VERSION.tar.gz", type="source", repos=NULL, lib="/opt/conda/lib/R/library")'             -e 'unlink("SparkR_$SPARKR_VERSION.tar.gz")'' returned a non-zero code: 1
make[1]: *** [../.image-demo-base] Error 1
gmake: *** [Makefile:164: demo-base] Error 2
```
but I can build an image from `main`.


Have you ensured that `make demo-base enterprise-gateway-demo itest-yarn` succeeds without pulling `demo-base:3.2.1` when building `enterprise-gateway-demo`?

File: etc/docker/demo-base/Dockerfile (hunk 1 above)
Comment (lresende) [reply etc/docker/demo-base/Dockerfile:69]: The mamba image has created a user with 1000 groupid as the jovyan user we use.

**Comment by lresende** [comment]
Something strange indeed, looks like with the new image, the RScript is not resolving the env vars embedded on the URL string but moving back to the miniconda3 image it works ok. So I updated the pr to use miniconda3 base image, but use mamba for general package management.

**Review by kevin-bates — APPROVED** [review]
LGTM - thanks @lresende.
